cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```diff
@@ -59,7 +109,9 @@ public CombinedHBaseMetricsTable(MetricsTable v2HBaseTable, MetricsTable v3HBase
 
   @Override
   public void put(SortedMap<byte[], ? extends SortedMap<byte[], Long>> updates) {
-    deleteColumns(updates);
+    if (tableExists.apply(resolution)) {
+      deleteColumns(updates);
```

Comment:
also, this does not protect you from exceptions, because the table may be disabled or get deleted between the check and the delete. Instead, you need to catch IOException from the v2Table, and if it does not exist any longer, remember that (for example, by setting v2Table to null) and not attempt to write to it again.